sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/FieldValueTests.cs

```diff
@@ -17,5 +18,88 @@ public void AsStringReturnsNullWhenFieldValueIsDefault()
             FieldValue value = default;
             Assert.IsNull(value.AsString());
         }
+
+        [Test]
+        public void InstantiateFieldValueWithFloatAndNoValueNumberAndText()
+        {
+            var fieldValueInternal = new FieldValue_internal(
+                FieldValueType.Float,
+                valueString: default,
+                valueDate: default,
+                valueTime: default,
+                valuePhoneNumber: default,
+                valueNumber: default,
+                valueInteger: default,
+                valueArray: default,
+                valueObject: default,
+                valueSelectionMark: default,
+                valueGender: default,
+                valueCountry: default,
+                text: "25.00%",
+                boundingBox: default,
+                confidence:  default,
+                elements: default,
+                page: default);
+
+            var fieldValue = new FieldValue(fieldValueInternal, default);
+
+            Assert.AreEqual(FieldValueType.Float, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsFloat());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithInt64AndNoValueInteger()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Int64);
+
+            Assert.AreEqual(FieldValueType.Int64, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsInt64());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithFloatAndNoValueNumber()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Float);
+
+            Assert.AreEqual(FieldValueType.Float, fieldValue.ValueType);
+            Assert.Throws<NullReferenceException>(() => fieldValue.AsFloat());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithDateAndNoValueDate()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Date);
+
+            Assert.AreEqual(FieldValueType.Date, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsDate());
+        }
+
+        [Test]
+        public void InstantiateFieldValueWithTimeAndNoValueTime()
+        {
+            var fieldValue = new FieldValue(FieldValueType.Time);
+
+            Assert.AreEqual(FieldValueType.Time, fieldValue.ValueType);
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsTime());
+        }
+
+        [Test]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/17814")]
+        public void InstantiateFieldValueWithSelectionMarkAndNoValueSelectionMark()
+        {
+            var fieldValue = new FieldValue(FieldValueType.SelectionMark);
+
+            Assert.AreEqual(FieldValueType.SelectionMark, fieldValue.ValueType);
+            Assert.AreEqual(SelectionMarkState.Selected, fieldValue.AsSelectionMarkState());
+        }
```

Comment:
I know we're ignoring this test for now, but it should be the same as others, shouldn't it?
```cs
Assert.AreEqual(FieldValueType.SelectionMark, fieldValue.ValueType);
Assert.Throws<InvalidOperationException>(() => fieldValue.AsSelectionMarkState());
```

Reply:
forgot to update after removing the `text` property